var/spack/repos/builtin/packages/libsm/package.py

```diff
@@ -16,10 +16,12 @@ class Libsm(AutotoolsPackage, XorgPackage):
     version('1.2.2', sha256='14bb7c669ce2b8ff712fbdbf48120e3742a77edcd5e025d6b3325ed30cf120f4')
 
     depends_on('libice@1.0.5:')
-    depends_on('libuuid')
+    if sys.platform != 'darwin':
+        # On macOS systems, Spack's libuuid conflicts with the system-installed
+        # version and breaks anything linked against Cocoa/Carbon.
+        depends_on('libuuid')
```

Comment:
Why not:
```suggestion
    # On macOS systems, Spack's libuuid conflicts with the system-installed
    # version and breaks anything linked against Cocoa/Carbon.
    depends_on('libuuid', when='platform=linux')
```
?

Reply:
What platforms are there aside from Mac and Linux? I don't think that's documented anywhere.